0x/Contracts/Exchange.cs

```diff
@@ -0,0 +1,151 @@
+﻿
+using Nethereum.Contracts;
+using Nethereum.Hex.HexTypes;
+using Nethereum.RPC.Eth.DTOs;
+using Nethereum.Web3;
+using System;
+using System.Collections.Generic;
+using System.Numerics;
+using System.Threading.Tasks;
+using ZeroX.Enums;
+using ZeroX.Orders;
+using ZeroX.Utilities;
+using Transaction = ZeroX.Transactions.Transaction;
+using Nethereum.Hex.HexConvertors.Extensions;
+
+namespace ZeroX.Contracts
+{
+    public class Exchange : SmartContract
+    {
+        // TODO: Add argument validation, add docs
+
+        static Exchange()
+        {
+            _abi = LoadABI(nameof(Exchange));
```

Comment:
If this method can throw, try to avoid it. Generally, having initializing methods in the constructor is a bad practice. When having these kind of situations (some initialization before using your object) stick to the `Factory` or `AbstractFactory` pattern. Easy to use, convenient and keeps your code clean.

Reply:
If this throws, it is a bug.